puppet/modules/web/files/deploy-yumrepo.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -xe
+# This is a forced SSH command - uncomment to test locally
+set -f -- $SSH_ORIGINAL_COMMAND
+
+prepcache() {
+	if [[ -e $REPO_PATH ]]; then
+		if [[ $MERGE == true ]] || [[ $OVERWRITE == true ]] ; then
+			echo "Repo overwrite (${OVERWRITE}) or merge (${MERGE}) are disabled, but ${REPO_PATH} already exists"
+			exit 1
+		fi
+		cp -al $REPO_PATH "$REPO_INSTANCE_PATH"
+	else
+		mkdir -p $REPO_INSTANCE_PATH
+	fi
+}
+
+do_rsync() {
+	opts=(--archive --verbose --hard-links --log-file "$REPO_RSYNC_LOG")
+	if [[ $MERGE == true ]] ; then
+		opts+=('--exclude=**/repodata/')
+	else
+		opts+=('--delete')
+	fi
+	rsync "${opts[@]}" --log-file-format 'CHANGED %f' "${REPO_SOURCE_RPM}/*" "${REPO_INSTANCE_PATH}/"
+	rsync "${opts[@]}" --log-file-format 'CHANGED source/%f' "${REPO_SOURCE_SRPM}/" "${REPO_INSTANCE_PATH}/source/"
+
+	set +f
+	for d in "${REPO_INSTANCE_PATH}"/*; do
+		(
+			cd "$d"
```

Comment:
pushd/popd?

Reply:
I'm not a fan of that. I've seen weird interactions when used with sudo where the user couldn't return to the previous directory. Changing directory in a subshell is what the previous script did and in my experience it works reliably.

Reply:
k